### PR TITLE
backend: (riscv) fix riscv_scf lowering

### PR DESCRIPTION
**Pull request by superlopuh**
I added a test for the lowering in Toy, and found a bug in our for loop lowering code. Hurray for tests :)

### CODE REVIEW

File: xdsl/backend/riscv/riscv_scf_to_asm.py [hunk 1]
```diff
@@ -53,7 +53,7 @@ def match_and_rewrite(self, op: riscv_scf.ForOp, rewriter: PatternRewriter, /):
         # upper bound and branching to the loop body if the condition is met.
         rewriter.insert_op_before_matched_op(
             [
-                get_loop_var := riscv.GetRegisterOp(loop_var_reg),
+                get_loop_var := riscv.MVOp(op.lb, rd=loop_var_reg),
```
Comment (superlopuh) [inline xdsl/backend/riscv/riscv_scf_to_asm.py:56]: When in a nested loop, the iteration index variable was never reset to the beginning of the loop.

**Review by nazavode — APPROVED** [review]
asm ftw

File: xdsl/backend/riscv/riscv_scf_to_asm.py (hunk 1 above)
Comment (tobiasgrosser) [reply xdsl/backend/riscv/riscv_scf_to_asm.py:56]: Should this not also update a filecheck test case?

Comment (superlopuh) [reply xdsl/backend/riscv/riscv_scf_to_asm.py:56]: yep, fixing

**Pull request merged**